### PR TITLE
Use linkerd-jaeger extension for control plane tracing

### PR DESCRIPTION
**Pull request by adleong**
Now that tracing has been split out of the main control plane and into the linkerd-jaeger extension, we remove references to tracing from the main control plane including:

* removing the tracing components from the main control plane chart
* removing the tracing injection logic from the main proxy injector and inject CLI (these will be added back into the new injector in the linkerd-jaeger extension)
* removing tracing related checks (these will be added back into `linkerd jaeger check`)
* removing related tests

We also update the `--control-plane-tracing` flag to configure the control plane components to send traces to the linkerd-jaeger extension.  To make sure this works even when the linkerd-jaeger extension is installed in a non-default namespace, we also add a `--control-plane-tracing-namespace` flag which can be used to change the namespace that the control plane components send traces to.

Note that for now, only the control plane components send traces; the proxies in the control plane do not.  This is because the linkerd-jaeger injector is not yet available.  However, this change adds the appropriate namespace annotations to the control plane namespace to configure the proxies to send traces to the linkerd-jaeger extension once the linkerd-jaeger injector is available.

I tested this by doing the following:

1. bin/linkerd install | kubectl apply -f -
1. bin/helm install jaeger jaeger/charts/jaeger
1. bin/linkerd upgrade --control-plane-tracing=true | kubectl apply -f -
1. kubectl -n linkerd-jaeger port-forward svc/jaeger 16686
1. open http://localhost:16686
1. see traces from the linkerd control plane

### CODE REVIEW

**Comment by Pothulapati** [comment]
> Note that for now, only the control plane components send traces; the proxies in the control plane do not. This is because the linkerd-jaeger injector is not yet available. However, this change adds the appropriate namespace annotations to the control plane namespace to configure the proxies to send traces to the linkerd-jaeger extension once the linkerd-jaeger injector is available.

As we merged the webhook recently, I tried doing this with steps that you mentioned but still could not see the `TRACE` env variables be set to the control plane proxies. Can you check this? 🤔

**Comment by adleong** [comment]
@Pothulapati good catch!  This was because I had missed removing the podinfo volume from the control plane components so the jaeger-injector was assuming they were already mutated and skipping them.  This is now fixed.

**Comment by Pothulapati** [comment]
This works great to me,

but after the upgrade, the injector does not seem to work well and fails. Any idea why?

Proxy Injector Logs
```
tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$!?⇣] via 🐹 v1.15.4
❯ k -n linkerd logs deploy/linkerd-proxy-injector proxy-injector
time="2020-12-04T09:26:20Z" level=info msg="running version dev-9a7a9837-tarun"
time="2020-12-04T09:26:20Z" level=info msg="waiting for caches to sync"
time="2020-12-04T09:26:21Z" level=info msg="caches synced"
time="2020-12-04T09:26:21Z" level=info msg="starting admin server on :9995"
time="2020-12-04T09:26:21Z" level=info msg="listening at :8443"
time="2020-12-04T09:34:56Z" level=info msg="received admission review request cc2a04a1-3d74-434a-8731-bcf17f2a1526"
time="2020-12-04T09:34:56Z" level=info msg="received pod/proxy-mutator-75cb9d5b5f-"
time="2020-12-04T09:34:56Z" level=info msg="patch generated for: pod/proxy-mutator-75cb9d5b5f-"
time="2020-12-04T09:34:56Z" level=info msg="received admission review request b626782a-4d15-4036-91c1-a4254717ea32"
time="2020-12-04T09:34:56Z" level=info msg="received pod/collector-69cc44dfbc-"
time="2020-12-04T09:34:56Z" level=info msg="patch generated for: pod/collector-69cc44dfbc-"
time="2020-12-04T09:34:56Z" level=info msg="received admission review request 7925af4f-e92e-494f-bda2-a732d72f46e1"
time="2020-12-04T09:34:56Z" level=info msg="received pod/jaeger-6f98d5c979-"
time="2020-12-04T09:34:56Z" level=info msg="patch generated for: pod/jaeger-6f98d5c979-"
2020/12/04 09:40:10 http: TLS handshake error from 127.0.0.1:41056: remote error: tls: bad certificate
2020/12/04 09:40:10 http: TLS handshake error from 127.0.0.1:41066: remote error: tls: bad certificate
2020/12/04 09:40:11 http: TLS handshake error from 127.0.0.1:41100: remote error: tls: bad certificate
2020/12/04 09:40:58 http: TLS handshake error from 127.0.0.1:42388: remote error: tls: bad certificate
2020/12/04 09:40:58 http: TLS handshake error from 127.0.0.1:42394: remote error: tls: bad certificate
2020/12/04 09:40:58 http: TLS handshake error from 127.0.0.1:42404: remote error: tls: bad certificate
2020/12/04 09:40:58 http: TLS handshake error from 127.0.0.1:42410: remote error: tls: bad certificate
```

File: test/integration/tracing/testdata/emojivoto.yaml
Comment (alpeb) [inline test/integration/tracing/testdata/emojivoto.yaml:35]: That's the default SA. Isn't it being picked up without this annotation?

**Comment by alpeb** [comment]
@Pothulapati I couldn't reproduce your issue. Could you provide repo steps?

**Comment by Pothulapati** [comment]
@alpeb Replication steps

```bash
tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
❯ k version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-14T07:30:52Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
❯ ./bin/linkerd install | k apply -f -
namespace/linkerd created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-identity created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-identity created
serviceaccount/linkerd-identity created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-controller created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-controller created
serviceaccount/linkerd-controller created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-destination created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-destination created
serviceaccount/linkerd-destination created
role.rbac.authorization.k8s.io/linkerd-heartbeat created
rolebinding.rbac.authorization.k8s.io/linkerd-heartbeat created
serviceaccount/linkerd-heartbeat created
role.rbac.authorization.k8s.io/linkerd-web created
rolebinding.rbac.authorization.k8s.io/linkerd-web created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-web-check created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-web-check created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-web-admin created
serviceaccount/linkerd-web created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/serviceprofiles.linkerd.io created
customresourcedefinition.apiextensions.k8s.io/trafficsplits.split.smi-spec.io created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-proxy-injector created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-proxy-injector created
serviceaccount/linkerd-proxy-injector created
secret/linkerd-proxy-injector-k8s-tls created
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
mutatingwebhookconfiguration.admissionregistration.k8s.io/linkerd-proxy-injector-webhook-config created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-sp-validator created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-sp-validator created
serviceaccount/linkerd-sp-validator created
secret/linkerd-sp-validator-k8s-tls created
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
validatingwebhookconfiguration.admissionregistration.k8s.io/linkerd-sp-validator-webhook-config created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-tap created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-tap-admin created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-tap created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-tap-auth-delegator created
serviceaccount/linkerd-tap created
rolebinding.rbac.authorization.k8s.io/linkerd-linkerd-tap-auth-reader created
secret/linkerd-tap-k8s-tls created
apiservice.apiregistration.k8s.io/v1alpha1.tap.linkerd.io created
podsecuritypolicy.policy/linkerd-linkerd-control-plane created
role.rbac.authorization.k8s.io/linkerd-psp created
rolebinding.rbac.authorization.k8s.io/linkerd-psp created
configmap/linkerd-config created
secret/linkerd-identity-issuer created
service/linkerd-identity created
service/linkerd-identity-headless created
deployment.apps/linkerd-identity created
service/linkerd-controller-api created
deployment.apps/linkerd-controller created
service/linkerd-dst created
service/linkerd-dst-headless created
deployment.apps/linkerd-destination created
cronjob.batch/linkerd-heartbeat created
service/linkerd-web created
deployment.apps/linkerd-web created
deployment.apps/linkerd-proxy-injector created
service/linkerd-proxy-injector created
service/linkerd-sp-validator created
deployment.apps/linkerd-sp-validator created
service/linkerd-tap created
deployment.apps/linkerd-tap created
serviceaccount/linkerd-grafana created
configmap/linkerd-grafana-config created
service/linkerd-grafana created
deployment.apps/linkerd-grafana created
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-prometheus created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-prometheus created
serviceaccount/linkerd-prometheus created
configmap/linkerd-prometheus-config created
service/linkerd-prometheus created
deployment.apps/linkerd-prometheus created
secret/linkerd-config-overrides created

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4 took 8s
❯ kgp -A
NAMESPACE            NAME                                         READY   STATUS    RESTARTS   AGE
kube-system          coredns-f9fd979d6-4nhqk                      1/1     Running   0          16m
kube-system          coredns-f9fd979d6-fv48v                      1/1     Running   0          16m
kube-system          etcd-kind-control-plane                      1/1     Running   0          16m
kube-system          kindnet-d956j                                1/1     Running   0          16m
kube-system          kube-apiserver-kind-control-plane            1/1     Running   0          16m
kube-system          kube-controller-manager-kind-control-plane   1/1     Running   1          16m
kube-system          kube-proxy-88gjq                             1/1     Running   0          16m
kube-system          kube-scheduler-kind-control-plane            1/1     Running   1          16m
linkerd              linkerd-controller-8569545d9f-gj946          2/2     Running   0          82s
linkerd              linkerd-destination-6bff4788fc-kqmtg         2/2     Running   0          82s
linkerd              linkerd-grafana-585654d86f-c65jl             2/2     Running   0          79s
linkerd              linkerd-identity-9f8f9f968-2kzzr             2/2     Running   0          82s
linkerd              linkerd-prometheus-597898dc77-jp8p5          1/2     Running   0          79s
linkerd              linkerd-proxy-injector-855bdbbc45-mj26t      2/2     Running   0          81s
linkerd              linkerd-sp-validator-555b59fff8-54jlr        2/2     Running   0          80s
linkerd              linkerd-tap-579b985d45-jpk5x                 2/2     Running   0          80s
linkerd              linkerd-web-7fc45c6484-nnmq8                 2/2     Running   0          81s
local-path-storage   local-path-provisioner-78776bfc44-2bscs      1/1     Running   1          16m

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
❯ ./bin/linkerd jaeger install | k apply -f -
namespace/linkerd-jaeger created
deployment.apps/proxy-mutator created
service/proxy-mutator created
serviceaccount/collector created
serviceaccount/jaeger created
clusterrole.rbac.authorization.k8s.io/linkerd-jaeger-linkerd-jaeger-proxy-mutator created
clusterrolebinding.rbac.authorization.k8s.io/linkerd-jaeger-linkerd-jaeger-proxy-mutator created
serviceaccount/proxy-mutator created
secret/proxy-mutator-k8s-tls created
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
mutatingwebhookconfiguration.admissionregistration.k8s.io/linkerd-proxy-mutator-webhook-config created
configmap/collector-config created
service/collector created
deployment.apps/collector created
service/jaeger created
deployment.apps/jaeger created

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
❯ kgp -A
NAMESPACE            NAME                                         READY   STATUS     RESTARTS   AGE
kube-system          coredns-f9fd979d6-4nhqk                      1/1     Running    0          16m
kube-system          coredns-f9fd979d6-fv48v                      1/1     Running    0          16m
kube-system          etcd-kind-control-plane                      1/1     Running    0          17m
kube-system          kindnet-d956j                                1/1     Running    0          16m
kube-system          kube-apiserver-kind-control-plane            1/1     Running    0          17m
kube-system          kube-controller-manager-kind-control-plane   1/1     Running    1          17m
kube-system          kube-proxy-88gjq                             1/1     Running    0          16m
kube-system          kube-scheduler-kind-control-plane            1/1     Running    1          17m
linkerd-jaeger       collector-69cc44dfbc-6hmvc                   0/2     Init:0/1   0          2s
linkerd-jaeger       jaeger-6f98d5c979-xhxgz                      0/2     Init:0/1   0          2s
linkerd-jaeger       proxy-mutator-6488659587-r92ww               0/2     Init:0/1   0          2s
linkerd              linkerd-controller-8569545d9f-gj946          2/2     Running    0          96s
linkerd              linkerd-destination-6bff4788fc-kqmtg         2/2     Running    0          96s
linkerd              linkerd-grafana-585654d86f-c65jl             2/2     Running    0          93s
linkerd              linkerd-identity-9f8f9f968-2kzzr             2/2     Running    0          96s
linkerd              linkerd-prometheus-597898dc77-jp8p5          2/2     Running    0          93s
linkerd              linkerd-proxy-injector-855bdbbc45-mj26t      2/2     Running    0          95s
linkerd              linkerd-sp-validator-555b59fff8-54jlr        2/2     Running    0          94s
linkerd              linkerd-tap-579b985d45-jpk5x                 2/2     Running    0          94s
linkerd              linkerd-web-7fc45c6484-nnmq8                 2/2     Running    0          95s
local-path-storage   local-path-provisioner-78776bfc44-2bscs      1/1     Running    1          16m

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
❯ ./bin/linkerd upgrade --control-plane-tracing=true | k apply -f -
namespace/linkerd unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-identity unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-identity unchanged
serviceaccount/linkerd-identity unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-controller unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-controller unchanged
serviceaccount/linkerd-controller unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-destination unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-destination unchanged
serviceaccount/linkerd-destination unchanged
role.rbac.authorization.k8s.io/linkerd-heartbeat unchanged
rolebinding.rbac.authorization.k8s.io/linkerd-heartbeat unchanged
serviceaccount/linkerd-heartbeat unchanged
role.rbac.authorization.k8s.io/linkerd-web unchanged
rolebinding.rbac.authorization.k8s.io/linkerd-web unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-web-check unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-web-check unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-web-admin unchanged
serviceaccount/linkerd-web unchanged
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/serviceprofiles.linkerd.io unchanged
customresourcedefinition.apiextensions.k8s.io/trafficsplits.split.smi-spec.io unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-proxy-injector unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-proxy-injector configured
serviceaccount/linkerd-proxy-injector unchanged
secret/linkerd-proxy-injector-k8s-tls configured
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
mutatingwebhookconfiguration.admissionregistration.k8s.io/linkerd-proxy-injector-webhook-config configured
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-sp-validator unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-sp-validator configured
serviceaccount/linkerd-sp-validator unchanged
secret/linkerd-sp-validator-k8s-tls configured
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
validatingwebhookconfiguration.admissionregistration.k8s.io/linkerd-sp-validator-webhook-config configured
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-tap unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-tap-admin unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-tap unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-tap-auth-delegator unchanged
serviceaccount/linkerd-tap unchanged
rolebinding.rbac.authorization.k8s.io/linkerd-linkerd-tap-auth-reader unchanged
secret/linkerd-tap-k8s-tls configured
apiservice.apiregistration.k8s.io/v1alpha1.tap.linkerd.io configured
podsecuritypolicy.policy/linkerd-linkerd-control-plane configured
role.rbac.authorization.k8s.io/linkerd-psp unchanged
rolebinding.rbac.authorization.k8s.io/linkerd-psp unchanged
configmap/linkerd-config configured
secret/linkerd-identity-issuer unchanged
service/linkerd-identity unchanged
service/linkerd-identity-headless unchanged
deployment.apps/linkerd-identity configured
service/linkerd-controller-api unchanged
deployment.apps/linkerd-controller configured
service/linkerd-dst unchanged
service/linkerd-dst-headless unchanged
deployment.apps/linkerd-destination configured
cronjob.batch/linkerd-heartbeat unchanged
service/linkerd-web unchanged
deployment.apps/linkerd-web configured
deployment.apps/linkerd-proxy-injector configured
service/linkerd-proxy-injector unchanged
service/linkerd-sp-validator unchanged
deployment.apps/linkerd-sp-validator configured
service/linkerd-tap unchanged
deployment.apps/linkerd-tap configured
serviceaccount/linkerd-grafana unchanged
configmap/linkerd-grafana-config unchanged
service/linkerd-grafana unchanged
deployment.apps/linkerd-grafana configured
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-prometheus unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-prometheus unchanged
serviceaccount/linkerd-prometheus unchanged
configmap/linkerd-prometheus-config unchanged
service/linkerd-prometheus unchanged
deployment.apps/linkerd-prometheus configured
secret/linkerd-config-overrides configured

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4 took 3s
❯

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4 took 3s
❯ kgp -A
NAMESPACE            NAME                                         READY   STATUS     RESTARTS   AGE
kube-system          coredns-f9fd979d6-4nhqk                      1/1     Running    0          17m
kube-system          coredns-f9fd979d6-fv48v                      1/1     Running    0          17m
kube-system          etcd-kind-control-plane                      1/1     Running    0          17m
kube-system          kindnet-d956j                                1/1     Running    0          17m
kube-system          kube-apiserver-kind-control-plane            1/1     Running    0          17m
kube-system          kube-controller-manager-kind-control-plane   1/1     Running    1          17m
kube-system          kube-proxy-88gjq                             1/1     Running    0          17m
kube-system          kube-scheduler-kind-control-plane            1/1     Running    1          17m
linkerd-jaeger       collector-69cc44dfbc-6hmvc                   2/2     Running    0          45s
linkerd-jaeger       jaeger-6f98d5c979-xhxgz                      2/2     Running    0          45s
linkerd-jaeger       proxy-mutator-6488659587-r92ww               2/2     Running    0          45s
linkerd              linkerd-controller-8569545d9f-gj946          2/2     Running    0          2m19s
linkerd              linkerd-controller-95cc944c-fxctm            0/2     Init:0/1   0          4s
linkerd              linkerd-destination-6565bf66b9-fxvdr         0/2     Init:0/1   0          4s
linkerd              linkerd-destination-6bff4788fc-kqmtg         2/2     Running    0          2m19s
linkerd              linkerd-grafana-585654d86f-c65jl             2/2     Running    0          2m16s
linkerd              linkerd-identity-66f75f445b-gxgcd            0/2     Init:0/1   0          4s
linkerd              linkerd-identity-9f8f9f968-2kzzr             2/2     Running    0          2m19s
linkerd              linkerd-prometheus-597898dc77-jp8p5          2/2     Running    0          2m16s
linkerd              linkerd-proxy-injector-855bdbbc45-mj26t      2/2     Running    0          2m18s
linkerd              linkerd-sp-validator-555b59fff8-54jlr        2/2     Running    0          2m17s
linkerd              linkerd-tap-579b985d45-jpk5x                 2/2     Running    0          2m17s
linkerd              linkerd-tap-854f6fbc5c-7w9zw                 0/2     Init:0/1   0          3s
linkerd              linkerd-web-7888cdb5d4-v7rwf                 0/2     Init:0/1   0          4s
linkerd              linkerd-web-7fc45c6484-nnmq8                 2/2     Running    0          2m18s
local-path-storage   local-path-provisioner-78776bfc44-2bscs      1/1     Running    1          17m

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
❯ k -n linkerd-jaeger port-forward svc/jaeger 16686
Forwarding from 127.0.0.1:16686 -> 16686
Forwarding from [::1]:16686 -> 16686
Handling connection for 16686
Handling connection for 16686
Handling connection for 16686
Handling connection for 16686
^C%

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4 took 32s
❯ ./bin/linkerd inject https://run.linkerd.io/emojivoto.yml | kubectl apply -f -

namespace "emojivoto" injected
serviceaccount "emoji" skipped
serviceaccount "voting" skipped
serviceaccount "web" skipped
service "emoji-svc" skipped
service "voting-svc" skipped
service "web-svc" skipped
deployment "emoji" injected
deployment "vote-bot" injected
deployment "voting" injected
deployment "web" injected

namespace/emojivoto created
serviceaccount/emoji created
serviceaccount/voting created
serviceaccount/web created
service/emoji-svc created
service/voting-svc created
service/web-svc created
deployment.apps/emoji created
deployment.apps/vote-bot created
deployment.apps/voting created
deployment.apps/web created

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4 took 2s
❯ k -n linkerd logs deploy/linkerd-proxy-injector proxy-injector
time="2020-12-04T17:30:44Z" level=info msg="running version git-00d205b0"
time="2020-12-04T17:30:44Z" level=info msg="waiting for caches to sync"
time="2020-12-04T17:30:44Z" level=info msg="caches synced"
time="2020-12-04T17:30:44Z" level=info msg="starting admin server on :9995"
time="2020-12-04T17:30:44Z" level=info msg="listening at :8443"
time="2020-12-04T17:31:55Z" level=info msg="received admission review request e3ab0ed8-51e7-4216-a3ff-c43affce13b8"
time="2020-12-04T17:31:55Z" level=info msg="received pod/proxy-mutator-6488659587-"
time="2020-12-04T17:31:55Z" level=info msg="patch generated for: pod/proxy-mutator-6488659587-"
time="2020-12-04T17:31:55Z" level=info msg="received admission review request 0427e025-63c2-4ce2-8d2c-eb16f2aae0d1"
time="2020-12-04T17:31:55Z" level=info msg="received pod/collector-69cc44dfbc-"
time="2020-12-04T17:31:55Z" level=info msg="patch generated for: pod/collector-69cc44dfbc-"
time="2020-12-04T17:31:55Z" level=info msg="received admission review request c16c51eb-65cb-4e60-a582-9a4be593c92b"
time="2020-12-04T17:31:55Z" level=info msg="received pod/jaeger-6f98d5c979-"
time="2020-12-04T17:31:55Z" level=info msg="patch generated for: pod/jaeger-6f98d5c979-"
2020/12/04 17:33:43 http: TLS handshake error from 127.0.0.1:57128: remote error: tls: bad certificate
2020/12/04 17:33:43 http: TLS handshake error from 127.0.0.1:57134: remote error: tls: bad certificate
2020/12/04 17:33:43 http: TLS handshake error from 127.0.0.1:57140: remote error: tls: bad certificate
2020/12/04 17:33:43 http: TLS handshake error from 127.0.0.1:57148: remote error: tls: bad certificate

tarun in dev in on  kind-kind () linkerd2 on  alex/gone-without-a-trace [$?] via 🐹 v1.15.4
```

**Comment by alpeb** [comment]
Thanks @Pothulapati , I think you've found a nasty bug, that I can reproduce in the latest edge :-(
Whenever upgrading linkerd to change anything, say something else innocuous like `--disable-heartbeat`, this will happen. Likely it's because the injector's mutatingwebhookconfiguration and the associated secret are getting regenerated during the upgrade, but the injector pod isn't getting restarted so it's not using the latest secret.

I've just merged the webhook dynamic SSL cert reload (#5282) into `main` and merging it in this branch solves the issue as the secret change is detected (note it takes a couple of minutes though). I'm not totally sure though if this requires further investigation...

**Comment by adleong** [comment]
This is a good find.  Let's not block this change on it, but we should definitely spend more time investigating this.  @alpeb do you happen to know why recovering takes a couple of minutes?  It would be ideal if the changed cert would be detected right away.

**Comment by alpeb** [comment]
> do you happen to know why recovering takes a couple of minutes? It would be ideal if the changed cert would be detected right away.

I think that's a k8s thing; just as configmaps, secrets changes take a while to be synced:
https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically

**Review by alpeb — APPROVED** [review]

File: charts/linkerd2/README.md
Comment (kleimkuhler) [inline charts/linkerd2/README.md:178]: Keep the description here?

```suggestion
| global.proxy.uid | int | `2102` | User id under which the proxy runs |
```

File: charts/linkerd2/README.md
Comment (kleimkuhler) [inline charts/linkerd2/README.md:143]: I know it's pretty clear what this is, but add a description here to stay consistent with everything else?

**Review by kleimkuhler — APPROVED** [review]
This looks good! I have a few non-blocking comments, but otherwise the change looks good.

**Comment by kleimkuhler** [comment]
By the way it looks like the `jaeger-webhook` image is not having it's version substituted correctly:

```
Normal   BackOff    3s (x4 over 54s)   kubelet, k3d-k3s-default-server-0  Back-off pulling image "ghcr.io/linkerd/jaeger-webhook:linkerdVersionValue"
```

**Comment by adleong** [comment]
That's.... weird.  It was working fine before this commit: https://github.com/linkerd/linkerd2/pull/5299/commits/cac1dcf4adaade421889e6286fa8ed0047988d0c and that seems totally unrelated to version substitution.

**Comment by adleong** [comment]
Ok I think the linkerd version replacement got broken in https://github.com/linkerd/linkerd2/pull/5304 but there aren't any tests for that until this PR which is why the error showed up here after merging main.  I've fixed the issue on this branch.

**Comment by kleimkuhler** [comment]
Yep good catch 👍

**Pull request merged**
